### PR TITLE
Fix failing integration tests

### PR DESCRIPTION
**Pull request by backsapc**
The integration tests became failing after merging https://github.com/warden-protocol/wardenprotocol/pull/729. This is a possible solution

## Summary by CodeRabbit

- **New Features**
	- Introduced a new pruning mechanism for actions, enhancing management based on timing parameters.
- **Improvements**
	- Updated the end block processing to conditionally execute pruning, improving control flow and action management.

### CODE REVIEW

File: warden/x/act/keeper/abci.go [hunk 1]
```diff
@@ -3,24 +3,44 @@ package keeper
...
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"time"
 )
 
 func (k Keeper) EndBlocker(ctx context.Context) error {
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
 	blockHeight := sdkCtx.BlockHeight()
 	params := k.GetParams(ctx)
 
+	if params.MaxPendingTime > 0 && params.MaxCompletedTime > 0 {
```
Comment (backsapc) [inline warden/x/act/keeper/abci.go:14]: @Pitasi, tbh, I’m not sure what the right way is to add parameters to a module that didn’t have them before. The issue is that the generated snapshots don’t store the parameters, so they’re being initialized with default (0) values. 
I’d appreciate it if you could give me some direction on how to set default values during initialization from the genesis.

Comment (pitasi) [reply warden/x/act/keeper/abci.go:14]: I see that `genesis.json` contains the default params correctly:

```json
    "act": {
      "params": {
        "max_pending_time": "86400s",
        "max_completed_time": "604800s",
        "prune_check_block_frequency": "10000"
      },
      "actions": [],
      "rules": []
    },
```

In `x/act/module/genesis.go` there is the `InitGenesis` function the is invoked the first time the chain starts, reading the genesis file. Inside, it calls `k.SetParams()` using the params contained in the genesis file.

Everything seems correct, in fact you can launch localnet and query the params:
```
$ wardend q act params
params:
  max_completed_time: 168h0m0s
  max_pending_time: 24h0m0s
  prune_check_block_frequency: "10000"
```

I'll continue investigating why it wasn't working before.

Comment (backsapc) [reply warden/x/act/keeper/abci.go:14]: Yes, but I haven't regenerated the snapshots before merging, so they don't have params filled in them
image

Comment (backsapc) [reply warden/x/act/keeper/abci.go:14]: https://github.com/warden-protocol/wardenprotocol/blob/main/tests/testdata/snapshot-base/config/genesis.json

**Review by pitasi — APPROVED** [review]
So, I think that the snapshots were generated using the wrong version of `wardend`.
In any case, we can merge this since it's a useful sanity check.

File: warden/x/act/keeper/abci.go (hunk 1 above)
Comment (pitasi) [reply warden/x/act/keeper/abci.go:14]: i'm gonna merge this, then in #800 i regenerated the snapshots correctly

**Pull request merged**